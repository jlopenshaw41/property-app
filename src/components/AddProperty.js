import React from "react";

const AddProperty = () => {
  return <div>Add property</div>;
};

export default AddProperty;
