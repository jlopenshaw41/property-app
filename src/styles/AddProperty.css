.AddProperty {
    padding: 10px;
}