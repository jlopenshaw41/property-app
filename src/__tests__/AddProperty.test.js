import React from "react";
import { render } from "@testing-library/react";
import AddProperty from "../components/AddProperty";

describe("Properties", () => {
  it("renders as expected", () => {
    const { asFragment } = render(<AddProperty />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a submit button", () => {
    const { getByTestId } = render(<AddProperty />);
    const submitButton = getByTestId("submit-button");
    expect(submitButton).toBeTruthy();
  });

  it("renders property title input", () => {
    const { getByTestId } = render(<AddProperty />);
    const propertyTitleInput = getByTestId("property-title-input");
    expect(propertyTitleInput).toBeTruthy();
  });

  it("renders type dropdown list", () => {
    const { getByTestId } = render(<AddProperty />);
    const typeDropdown = getByTestId("type-dropdown");
    expect(typeDropdown).toBeTruthy();
  });

  it("renders city dropdown list", () => {
    const { getByTestId } = render(<AddProperty />);
    const cityDropdown = getByTestId("city-dropdown");
    expect(cityDropdown).toBeTruthy();
  });
});
