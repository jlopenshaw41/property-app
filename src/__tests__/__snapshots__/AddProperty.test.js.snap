// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProperty renders as expected 1`] = `
<DocumentFragment>
  <div
    class="AddProperty"
  >
    <div
      class="Alert"
    >
      Error!
    </div>
    <form>
      <div>
        <label
          for="title"
        >
          Property title:
          <input
            data-testid="property-title-input"
            id="title"
            name="title"
            placeholder="e.g. 5 bedroom detached house for sale"
            type="text"
            value=""
          />
        </label>
      </div>
      <div>
        <label
          for="type"
        >
          Property type:
          <select
            data-testid="type-dropdown"
            id="type"
            name="type"
          >
            <option
              value="Flat"
            >
              Flat
            </option>
            <option
              value="Detached"
            >
              Detached
            </option>
            <option
              value="Semi-detached"
            >
              Semi-detached
            </option>
            <option
              value="Terraced"
            >
              Terraced
            </option>
            <option
              value="End of terrace"
            >
              End of terrace
            </option>
            <option
              value="Cottage"
            >
              Cottage
            </option>
            <option
              value="Bungalow"
            >
              Bungalow
            </option>
          </select>
        </label>
      </div>
      <div>
        <label
          for="bedrooms"
        >
          Bedrooms:
          <input
            data-testid="bedrooms-input"
            id="bedrooms"
            min="0"
            name="bedrooms"
            placeholder="0"
            type="number"
            value="0"
          />
        </label>
      </div>
      <div>
        <label
          for="bathrooms"
        >
          Bathrooms:
          <input
            data-testid="bathrooms-input"
            id="bathrooms"
            min="0"
            name="bathrooms"
            placeholder="0"
            type="number"
            value="0"
          />
        </label>
      </div>
      <div>
        <label
          for="price"
        >
          Price (Â£):
          <input
            data-testid="price-input"
            id="price"
            min="0"
            name="price"
            placeholder="0"
            type="number"
            value="0"
          />
        </label>
      </div>
      <div>
        <label
          for="city"
        >
          City:
          <select
            data-testid="city-dropdown"
            id="city"
            name="city"
          >
            <option
              value="Manchester"
            >
              Manchester
            </option>
            <option
              value="Leeds"
            >
              Leeds
            </option>
            <option
              value="Sheffield"
            >
              Sheffield
            </option>
            <option
              value="Liverpool"
            >
              Liverpool
            </option>
          </select>
        </label>
      </div>
      <div>
        <label
          for="email"
        >
          Contact email:
          <input
            data-testid="email-input"
            id="email"
            name="email"
            placeholder="e.g. name@email.co.uk"
            type="email"
            value=""
          />
        </label>
      </div>
      <button
        data-testid="submit-button"
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
